modules/auxiliary/scanner/smb/smb_enumshares.rb

```diff
@@ -83,7 +85,10 @@ def enum_tree(tree, share, subdir = '')
     write = tree.permissions.write_ea == 1
     skip = false
 
-    if ENUMERABLE_SHARE_TYPES.include?(share[:type])
+    # If the `Share` option has been set, then skip as we don't report on shares when working with a single share
+    if !datastore['Share'].nil?
+      skip
```

Comment:
Is there meant to be additional code here? 👀 

From a quick glance it looks like `skip` isn't a function, it's just a local variable reference? 🔍

Reply:
I had added this to test the functionality of skipping that entire `if` statement if we had a value for `datastore['Share']`. I never went back to update it. My bad 🙃 

Wrapped the original `share[:type]` logic in an `if` to only run if there is no value for `datastore['Store`].